### DISCUSSION TITLE
Debugging

### DISCUSSION THREAD

**Discussion by felipebueno** [stackernews/stacker.news#978] [local dev help]
I like exploring code bases using a visual debugger, setting breakpoints here and there and clicking things to see how they interact. Off course I also use debuggers **a lot** to debug :grin: things.

I'd like to configure my local sn environment to be able to debug using vscode or any other editor/ide but I have no experience with Next.js or any other full-stack js framework, specially inside docker containers. Do you (@huumn @ekzyis and the other devs) have any idea if this is possible? 

PS: I'll keep researching and I'll let you know if I find anything useful

**Comment by felipebueno** [comment]
Update: I know it's possible to debug things inside docker containers (I already did that with Laravel). I just don't know how to accomplish that with Next.js yet

**Reply by felipebueno** [reply]
I just saw there's a ```launch.json``` file in the project but nothing happens when I click "Launch Program"

![image](https://github.com/stackernews/stacker.news/assets/241271/4cd5296c-2ded-4b7b-be06-e7b131a5df04)

**Comment by huumn** [comment]
Nextjs has a guide here: https://nextjs.org/docs/pages/building-your-application/configuring/debugging

I tried it once a couple years ago, but I couldn't get it to work. It's had an update so it might work now.

@SatsAllDay was working on it awhile ago and ran into a bug: https://github.com/stackernews/stacker.news/pull/450

**Reply by felipebueno** [reply]
Nice! I'll take a look at the guide

**Reply by felipebueno** [reply]
> It's had an update so it might work now

Hmm it seems that's still ongoing. They just opened a new PR

https://github.com/vercel/next.js/pull/60919

![image](https://github.com/stackernews/stacker.news/assets/241271/e6ab3c17-db55-4daa-b460-b8c4fe63f7ab)

**Reply by SatsAllDay** [reply]
Yea it's not advancing as one would hope
